utils/duration.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+// Duration is a custom duration type that supports marshalling/unmarshalling.
+// This type and supporting functionality can be removed once go supports for
+// [time.Duration], which is planned for go2: https://github.com/golang/go/issues/10275
```

Comment:
That's a pretty unfortunate oversight from Golang. So we need JSON (un)marshaling in addition to proto (un)marshaling because we want to specify durations in JSON? Is this the first JSON duration field we've ever had? I guess the other "duration" settings we have are through environment variables (`VIAM_RESOURCE_CONFIGURATION_TIMEOUT` + `VIAM_MODULE_STARTUP_TIMEOUT`). I'm assuming we don't want to use an env variable for `first_run_timeout`?

Reply:
this is still WIP, so TBD on whether this ends up being ultimate solution. we handle other config duration fields (e.g. session heartbeat window) by maintaining a separate struct that has a "string" field for each duration field. this intermediate struct is validated and at that point we convert said strings to [time.Duration]s.

Reply:
i ended up successfully implementing the custom duration that supports marshaling/unmarshaling - i think using this new field is the approach i would prefer